allennlp/data/vocabulary.py

```diff
@@ -443,7 +444,7 @@ def set_from_file(
             self._token_to_index[namespace] = {}
             self._index_to_token[namespace] = {}
         with codecs.open(filename, "r", "utf-8") as input_file:
-            lines = input_file.read().split("\n")
+            lines = re.split(r"\n|\r\n?", input_file.read())
```

Comment:
This should be more efficient if we pre-compile the regex. So somewhere at the top of this module you could have 

```python
_NEW_LINE_REGEX = re.compile(r"\n|\r\n")
```

And then change this line to

```suggestion
            lines = _NEW_LINE_REGEX.split(input_file.read())
```

Reply:
Done!